Backend/Tournament_Public/middlewares/errorHandler.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Middleware to handle errors.
+ * 
+ * @param {Error} err - The error object.
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @param {Function} next - The next middleware function.
+ */
+
+// Error-handling middleware for Express
+
+const errorHandler = (err, res,) => {
+    console.error(err.stack);  
+
+    if (err.sqlState === '45000') {
+        return res.status(400).json({ message: err.message });
```

Comment:
not sure if this shld be 400 Bad Request? cause if missing request entry then the proc will just return null. 45000 highly likely is 500 Internal Server Error instead

Reply:
Regarding middleware think can ref jasper's ah he got gc1 approval also